src/vm_pmu.rs

```diff
@@ -368,4 +375,195 @@ impl VmPmuState {
             }
         }
     }
+
+    /// Helper function to call either SBI, or the internal implementation
+    /// for configure_matching_counters() depending on whether Sscdeleg is implemented
+    pub fn configure_matching_counters(
+        &self,
+        counter_index: u64,
+        counter_mask: u64,
+        config_flags: PmuCounterConfigFlags,
+        event_type: PmuEventType,
+        event_data: u64,
+    ) -> SbiResult<u64> {
+        if !CpuInfo::get().has_sscdeleg() {
+            sbi::api::pmu::configure_matching_counters(
+                counter_index,
+                counter_mask,
+                config_flags,
+                event_type,
+                event_data,
+            )
+        } else {
+            fn configure_counter_csrs(
+                config_flags: PmuCounterConfigFlags,
+                event_type: PmuEventType,
+                event_data: u64,
+                counter_index: u64,
+            ) {
+                let mut shpmevent = LocalRegisterCopy::<u64, shpmevent::Register>::new(0);
+                // Ignore vuinh / vsinh from the VM and translate uinh
+                if config_flags.is_uinh() {
+                    shpmevent.modify(shpmevent::vuinh.val(1));
+                }
+                if config_flags.is_sinh() {
+                    shpmevent.modify(shpmevent::sinh.val(1));
```

Comment:
This should be `vsinh`. And we should always be setting `sinh` to inhibit counting in Salus.

Reply:
Changed. As discussed offline, Salus was unconditionally setting sinh upstream, but it's been modified now.